main.tf

```
@@ -123,12 +123,17 @@ resource "azurerm_windows_function_app" "this" {
 }
 
 resource "azurerm_monitor_diagnostic_setting" "this" {
-  name                       = "audit-logs"
-  target_resource_id         = local.function_app.id
-  log_analytics_workspace_id = var.log_analytics_workspace_id
+  name                           = "audit-logs"
```

You should get this value from the variable you created:

```suggestion
  name                           = var.diagnostic_setting_name
```